### DISCUSSION TITLE
ListField with allow_empty=True should set required=False by default

### DISCUSSION THREAD

**Discussion by Winspear** [encode/django-rest-framework#9046] [Ideas & Suggestions]
Similar to how CharField with blank=True toggles required=False without explicit declaration.

**Comment by auvipy** [comment]
i would like to know more detail, you you elaborate please? with versions and examples

**Reply by Winspear** [reply]
Using djangorestframework==3.14.0, you can create a serializer like so
```
class MySerializer(BaseSerializer):
    a_required_char_field = serializers.CharField()
    a_not_required_char_field = serializers.CharField(blank=True)
    a_list_field = serializers.ListField(allow_empty=True)
```
If you then attempt to serialize an object with this serializer, it will raise a key error if it missing "a_required_char_field", it will *not* raise a key error if it is missing "a_not_required_char_field" (as setting blank=True automatically sets required=False for CharField), and it will raise a key error if a_list_field is missing. Given rest framework is already setting required=False flags in some cases without explicit flags being set, I believe ListField (and possibly other fields) should follow this pattern such that when setting allow_empty (or in the case of other fields, semantically similar flags), required is set to False.
Implementing this change however would be breaking for any users who were expecting their serializer to require an empty list.

**Reply by auvipy** [reply]
I will revisit this after v3.15 release
